drivers/include/at.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_at AT (Hayes) command set library
+ * @ingroup     drivers
+ * @brief       AT (Hayes) command set library
+ *
+ * This module provides functions to interact with devices using AT commands.
+ *
+ * Most functions compare the bytes echoed by the device with what they
+ * intended to send, and bail out if there's no match.
+ *
+ * Furthermore, the library tries to cope with difficulties regarding different
+ * line endings. It usually sends "<command><CR>", but expects
+ * "<command>\LF\CR" as echo.
+ *
+ * As a debugging aid, when compiled with "-DAT_PRINT_INCOMING=1", every input
+ * byte gets printed.
+ * @{
+ *
+ * @file
+ *
+ * @brief       AT (Hayes) library interface
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef AT_H
+#define AT_H
+
+#include <stdint.h>
+
+#include "isrpipe.h"
+#include "periph/uart.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef AT_END_OF_LINE
+/** End of line character to send after the AT command */
+#define AT_END_OF_LINE "\r"
+#endif
+
+/**
+ * @brief AT device structure
+ */
+typedef struct {
+    isrpipe_t isrpipe;      /**< isrpipe used for getting data from uart */
+    uart_t uart;            /**< UART device where the AT device is attached */
+} at_dev_t;
+
+/**
+ * @brief   Initialize AT device struct
+ *
+ * @param[in]   dev         struct to initialize
+ * @param[in]   uart        UART the device is connected to
+ * @param[in]   baudrate    baudrate of the device
+ * @param[in]   buf         input buffer
+ * @param[in]   bufsize     size of @p buf
+ *
+ * @returns     0 on success
+ * @returns     <0 otherwise
```

Comment:
Using an anonymous enum for return codes would improve the API of this library.

Reply:
not in this PR